eng/pipelines/common/templates/runtimes/run-test-job.yml

```diff
@@ -603,3 +648,74 @@ jobs:
         artifactName: '${{ parameters.runtimeFlavor }}_${{ parameters.runtimeVariant }}_$(LogNamePrefix)_$(osGroup)$(osSubgroup)_$(archType)_$(buildConfig)_${{ parameters.testGroup }}'
       continueOnError: true
       condition: always()
+
+    ########################################################################################################
+    #
+    # Finalize SuperPMI collection: (1) merge all MCH files generated by all Helix jobs, (2) upload MCH file to Azure Storage, (3) upload log files
+    # Note that all these steps are "condition: always()" because we want to upload as much of the collection
+    # as possible, even if there were test failures.
+    #
+    ########################################################################################################
+
+    - ${{ if eq(parameters.SuperPmiCollect, true) }}:
+
+      # Create required directories for merged mch collection and superpmi logs
+      - ${{ if ne(parameters.osGroup, 'windows') }}:
```

Comment:
There are about 50 lines that are very similar. If this was all in C#, I might suggest sharing via some refactoring. However, this is via very complex YML templating, so it's no clear to me if addition more complexity to enable sharing is warranted. E.g., perhaps a new YML template could be defined that is invoked in both places. It might be worth considering as a follow-up step, for instance, when adding similar logic to support run of the libraries tests as well.

Reply:
Yes, I am just afraid that we might update things in future at one place but not other. I agree, lets keep it as is for now and think if we run into such problems.